Tella/Data/Networking/Repositories/GDriveRepository.swift

```diff
@@ -0,0 +1,450 @@
+            if error != nil {
+                handleSubmissionError(uploadProgressInfo: uploadProgressInfo, promise: promise)
+                return
+            }
+            
+            guard file is GTLRDrive_File else {
+                handleSubmissionError(uploadProgressInfo: uploadProgressInfo, promise: promise)
+                return
+            }
+            
+            uploadProgressInfo.bytesSent = Int(uploadProgressInfo.total!)
+            uploadProgressInfo.current = Int(uploadProgressInfo.total!)
+            uploadProgressInfo.status = .uploaded
+            promise(.success(uploadProgressInfo))
+        }
+        
+        self.uploadTasks[fileId] = ticket
+    }
+    
+    private func checkFileExists(fileName: String, folderId: String, driveService: GTLRDriveService) async throws -> Bool {
+        return try await withCheckedThrowingContinuation { continuation in
+            let query = GTLRDriveQuery_FilesList.query()
+            query.q = "name = '\(fileName)' and '\(folderId)' in parents and trashed = false"
+            query.fields = "files(id, name)"
+            query.includeItemsFromAllDrives = true
+            query.supportsAllDrives = true
+            
+            driveService.executeQuery(query) { (_, response, error) in
+                if let error = error {
+                    continuation.resume(throwing: error)
+                    return
+                }
+                
+                if let fileList = response as? GTLRDrive_FileList, let files = fileList.files, !files.isEmpty {
+                    continuation.resume(returning: true)
+                } else {
+                    continuation.resume(returning: false)
+                }
+            }
+        }
+    }
+    
+    func pauseAllUploads() {
+        uploadLock.lock()
+        defer { uploadLock.unlock() }
+        
+        isUploading = false
+        uploadTasks.forEach { $0.value.cancel() }
+        uploadTasks.removeAll()
+    }
+
+    func resumeAllUploads() {
+        uploadLock.lock()
+        defer { uploadLock.unlock() }
+        
+        isUploading = true
+    }
+    
+    func signOut() {
+        GIDSignIn.sharedInstance.signOut()
+    }
+    
+    private func handleSubmissionError(uploadProgressInfo: UploadProgressInfo, promise: @escaping (Result<UploadProgressInfo, APIError>) -> Void) {
+        uploadProgressInfo.status = .submissionError
+        if !networkMonitor.isConnected {
+            uploadProgressInfo.error = APIError.noInternetConnection
+            promise(.failure(.noInternetConnection))
+            return
+        }
+        uploadProgressInfo.error = APIError.unexpectedResponse
+        promise(.failure(.unexpectedResponse))
+    }
+    
+    private func mapToAPIError(_ error: Error) -> APIError {
+        if let apiError = error as? APIError {
+            return apiError
+        }
+        
+        // Map other error types to APIError
+        if let nsError = error as NSError? {
```

Comment:
I think we can handle these error codes in `APIError`.httpCode(code)
if you see that the gDrive errors are different from the standard errors you can add .`gDrivehttpCode(code)` for example 
@Rimktarii